### ISSUE TITLE
Text blocks: Caret displayed in wrong position on line breaking

### ISSUE THREAD

**Issue by fluiddot** [WordPress/gutenberg#36526]
## Description

I noticed that in some cases related to how ~~the Heading block~~ text blocks (like Heading or Paragraph blocks) break lines when adding text, that the caret is displayed in the wrong position.

## Step-by-step reproduction instructions

1. Add a text block like the Heading or Paragraph block.
2. Add several words and try to make the last word to be long enough so the line is broken and the word goes to a second line.
3. Move the caret to the end of the first line.
4. Observe that the caret is displayed in the second line instead of in the first one.

## Expected behaviour

The caret should be displayed in the correct position.

## Actual behaviour

The caret is displayed in the wrong position.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/14905380/142029646-921d9899-eadf-4524-82fd-4b2d8a189b0c.MP4

## WordPress information
- WordPress version: N/A
- Gutenberg version: N/A
- Are all plugins except Gutenberg deactivated? N/A
- Are you using a default theme (e.g. Twenty Twenty-One)? N/A

## Device information
- Device: iPhone 11
- Operating system: iOS 15.1
- WordPress app version: `18.7`

**Comment by hypest** [comment]
I can reproduce on paragraph blocks as well, do you confirm Carlos?

**Comment by fluiddot** [comment]
> I can reproduce on paragraph blocks as well, do you confirm Carlos?

Yes, I confirm it, I've just verified that it also happens in the Paragraph block too. Looks like the issue is affecting text blocks in general, probably the ones that use the `RichText` component. I'll update the issue's description to reflect this information, thanks for pointing it out 🙇 .

**Comment by tyxla** [comment]
This is no longer relevant after the React Native implementation was removed in #78747. Closing.
